music_assistant/providers/nugs/__init__.py

```diff
@@ -0,0 +1,481 @@
+        return self._parse_playlist(response["items"])
+
+    async def get_album_tracks(self, prov_album_id) -> list[Track]:
+        """Get all album tracks for given album id."""
+        endpoint = f"shows/{prov_album_id}"
+        response = await self._get_data("catalog", endpoint)
+        album_data = response["Response"]
+        artist = await self.get_artist(album_data["artistID"])
+        album = self._get_item_mapping(
+            MediaType.ALBUM, album_data["containerID"], album_data["containerInfo"]
+        )
+        image = f"https://api.livedownloads.com{album_data['img']['url']}"
+        return [
+            self._parse_track(item, artist=artist, album=album, image=image)
+            for item in album_data["tracks"]
+            if item["trackID"]
+        ]
+
+    async def get_playlist_tracks(self, prov_playlist_id: str, page: int = 0) -> list[Track]:
+        """Get playlist tracks."""
+        result: list[Track] = []
+        if page > 0:
+            # paging not yet supported
+            return []
+        endpoint = f"/playlists/{prov_playlist_id}/playlist-tracks/all"
+        nugs_result = await self._get_data("stash", endpoint)
+        for index, item in enumerate(nugs_result["items"], 1):
+            track = self._parse_track(item)
+            track.position = index
+            result.append(track)
+        return result
+
+    async def get_stream_details(self, item_id: str, media_type: MediaType) -> StreamDetails:
+        """Return the content details for the given track when it will be streamed."""
+        stream_url = await self._get_stream_url(item_id)
+        return StreamDetails(
+            item_id=item_id,
+            provider=self.lookup_key,
+            audio_format=AudioFormat(
+                content_type=ContentType.UNKNOWN,
+            ),
+            stream_type=StreamType.HTTP,
+            path=stream_url,
+        )
+
+    def _parse_artist(self, artist_obj):
+        """Parse nugs artist object to generic layout."""
+        artist_id = artist_obj.get("artistID") or artist_obj.get("id")
+        artist_name = artist_obj.get("artistName") or artist_obj.get("name")
+        artist = Artist(
+            item_id=artist_id,
+            provider=self.lookup_key,
+            name=artist_name,
+            provider_mappings={
+                ProviderMapping(
+                    item_id=artist_id,
+                    provider_domain=self.domain,
+                    provider_instance=self.instance_id,
+                    url=f"https://catalog.nugs.net/api/v1/artists?ids={artist_id}",
+                )
+            },
+        )
+        if artist_obj.get("avatarImage"):
+            artist.metadata.images = [
+                MediaItemImage(
+                    type=ImageType.THUMB,
+                    path=artist_obj["avatarImage"]["url"],
+                    provider=self.lookup_key,
+                    remotely_accessible=True,
+                )
+            ]
```

Comment:
If you want to shorten this a tiny bit, you can also use the "add_image" helper on the metadata object.

artist.metadata.add_image(
    MediaItemImage(
                    type=ImageType.THUMB,
                    path=artist_obj["avatarImage"]["url"],
                    provider=self.lookup_key,
                    remotely_accessible=True,
                ))


Just for future reference, no need to fix now.